### PR TITLE
feat: impl datatypes2/timestamp

### PR DESCRIPTION
**Pull request by v0y4g3r**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
Impl timestamp datatypes and vectors in datatypes2

## Checklist

- [X]  I have written the necessary rustdoc comments.
- []  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/datatypes2/src/types/timestamp_type.rs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    TimestampNanosecondVectorBuilder, TimestampSecondVector, TimestampSecondVectorBuilder,
+};
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[enum_dispatch(DataType)]
+pub enum TimestampType {
+    TimestampSecondType(TimestampSecondType),
+    TimestampMillisecondType(TimestampMillisecondType),
+    TimestampMicrosecondType(TimestampMicrosecondType),
+    TimestampNanosecondType(TimestampNanosecondType),
```
Comment (evenyag) [inline src/datatypes2/src/types/timestamp_type.rs:50]: What about using more concise variant names?
```suggestion
    Second(TimestampSecondType),
    Millisecond(TimestampMillisecondType),
    Microsecond(TimestampMicrosecondType),
    Nanosecond(TimestampNanosecondType),
```

File: src/datatypes2/src/timestamp.rs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+use crate::types::{
+    TimestampMicrosecondType, TimestampMillisecondType, TimestampNanosecondType,
+    TimestampSecondType, WrapperType,
+};
+use crate::vectors::{
+    TimestampMicrosecondVector, TimestampMillisecondVector, TimestampNanosecondVector,
+    TimestampSecondVector,
+};
+
+macro_rules! define_timestamp_with_unit {
```
Comment (waynexia) [inline src/datatypes2/src/timestamp.rs:30]: AmAzing

**Review by waynexia — APPROVED** [review]

File: src/datatypes2/src/types/timestamp_type.rs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                fn as_arrow_type(&self) -> ArrowDataType {
+                    ArrowDataType::Timestamp(ArrowTimeUnit::$unit, None)
+                }
+
+                fn create_mutable_vector(&self, capacity: usize) -> Box<dyn MutableVector> {
+                    Box::new([<Timestamp $unit Vector Builder>]::with_capacity(capacity))
+                }
+            }
+
+
```
Comment (waynexia) [inline src/datatypes2/src/types/timestamp_type.rs:81]:
```suggestion
```

File: src/datatypes2/src/data_type.rs [hunk 4]
```diff
@@ -195,36 +195,56 @@ impl_new_concrete_type_functions!(
...
+    //                 Self::timestamp_datatype(TimeUnit::Millisecond)
+    //             }
+    //             arrow::datatypes::TimeUnit::Microsecond => {
+    //                 Self::timestamp_datatype(TimeUnit::Microsecond)
+    //             }
+    //             arrow::datatypes::TimeUnit::Nanosecond => {
+    //                 Self::timestamp_datatype(TimeUnit::Nanosecond)
+    //             }
+    //         }
+    //     }
```
[the remark this answers is not in the document]
Comment (v0y4g3r) [reply src/datatypes2/src/data_type.rs:238]: I think we can simply delete this.

**Review by waynexia — APPROVED** [review]
L6TM

**Review by evenyag — APPROVED** [review]
LGTM

**Pull request merged**